application/app/utils/GlobalHttpErrorHandler.java

```diff
@@ -0,0 +1,73 @@
+package utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.typesafe.config.Config;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.Environment;
+import play.api.OptionalSourceMapper;
+import play.api.UsefulException;
+import play.api.routing.Router;
+import play.http.DefaultHttpErrorHandler;
+import play.libs.Json;
+import play.mvc.Http.RequestHeader;
+import play.mvc.Result;
+import play.mvc.Results;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+
+/**
+ * Error Handler ensuring that errors in JSON endpoints are reported using a JSON result.
+ */
+@Singleton
+public class GlobalHttpErrorHandler extends DefaultHttpErrorHandler {
+    private static final Logger logger = LoggerFactory.getLogger(GlobalHttpErrorHandler.class);
+    private static final List<String> JSON_ENDPOINTS = Collections.singletonList("/api");
```

Comment:
This looks good as a standard content type definition. What happens if the client sends a different accepts content header like `application/xml`? Does it still return a json message?

Reply:
I have to read this: https://www.playframework.com/documentation/2.7.x/JavaErrorHandling
I guess there is a better solution than the GlobalHttpErrorHandler.

Reply:
You are right, the `play.http.HtmlOrJsonHttpErrorHandler` or a custom implementation alike could be suitable for us :+1: